src/DSL/Builder.cs

```diff
@@ -0,0 +1,307 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+//
+
+using AutoRest.Core.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace AutoRest.TypeScript.DSL
+{
+    public class Builder : IBuilder
+    {
+        private const string newLine = "\n";
+        private const string singleIndent = "  ";
+
+        private readonly StringBuilder contents = new StringBuilder();
```

Comment:
Instead of using `StringBuilder` we should use generators (`IEnumerable` and `yield return`) but I'm afraid it's a bigger task. `IEnumerable` should speed up the generation and reduce memory consumption.